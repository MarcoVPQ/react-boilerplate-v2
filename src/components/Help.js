import React from  'react';

const helpExpensePage = ()=>(
    <div>
        This is Help component
    </div>
)

export default helpExpensePage;