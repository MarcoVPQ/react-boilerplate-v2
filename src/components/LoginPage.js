import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from './../actions/aut';

export const LoginPage = ({startLogin}) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="layout__title">Boiler plate</h1>
            <p>Tagline for App</p>
            <button className="button" onClick={startLogin}>Login with Google</button>
        </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)