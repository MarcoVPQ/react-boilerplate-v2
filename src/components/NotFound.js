import React from 'react';
import { Link } from 'react-router-dom';


const Notfound = ()=>(
    <div>
        404!!- <Link to='/'>Go home</Link>
    </div>
);

export default Notfound;