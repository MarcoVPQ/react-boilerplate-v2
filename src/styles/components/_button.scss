.button{
    background: $bright-blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button__remove{
    background: #888888;
}

.button--logout{
    background: none;
}