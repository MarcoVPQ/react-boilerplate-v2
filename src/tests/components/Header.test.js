import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

let startLogout, wrapper;

beforeEach(() => {
    startLogout = jest.fn()
    wrapper = shallow(<Header startLogout={startLogout} />);
})

test('it should render header component', () => {
    expect(wrapper).toMatchSnapshot();   
})
test('Should call startLogout' , () => {
    wrapper.find('button').simulate('click')
    expect(startLogout).toHaveBeenCalled()
})