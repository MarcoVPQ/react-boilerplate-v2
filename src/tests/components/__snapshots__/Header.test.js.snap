// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render header component 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boiler plate
        </h1>
      </Link>
      <button
        className="button button--logout"
        onClick={[Function]}
      >
        Log out
      </button>
    </div>
  </div>
</header>
`;
